features/step_definitions/affordability.rb

```diff
@@ -284,55 +284,24 @@
 end
 
 Then(/^I should see the correct results for a single applicant$/) do
-  expect(step_three.repayments).to have_text('£1,706.21')
-  expect(step_three.risk_chart).to have_text('53%')
-  expect(step_three.essentials).to have_text('53%')
-  expect(step_three.essentials).to have_text('£3,206.21')
-  expect(step_three.total_leftover).to have_text('47%')
-  expect(step_three.total_leftover).to have_text('£2,793.79')
-  expect(step_three.remaining).to have_text('£1,893.79')
-  expect(step_three.what_if_changes).to have_text('2%')
-  expect(step_three.what_if_changes).to have_text('£2,103.35')
-  expect(step_three.what_if_remaining).to have_text('£1,496.65')
-end
-
-Then(/^I should see the correct results for a single applicant for JavaScript$/) do
-  expect(step_three.repayments_js).to have_text('£1,706.21')
-  expect(step_three.risk_chart_js).to have_text('53%')
-  expect(step_three.essentials_js).to have_text('53%')
-  expect(step_three.essentials_js).to have_text('£3,206.21')
-  expect(step_three.total_leftover_js).to have_text('47%')
-  expect(step_three.total_leftover_js).to have_text('£2,793.79')
-  expect(step_three.remaining_js).to have_text('£1,893.79')
-  expect(step_three.what_if_changes_js).to have_text('3%')
-  expect(step_three.what_if_changes_js).to have_text('£2,318.20')
-  expect(step_three.what_if_remaining_js).to have_text('£1,281.80')
-end
-
+  expect(js_test? ? step_three.repayments_js : step_three.repayments).to have_text('£1,706.21')
+  expect(js_test? ? step_three.risk_chart_js : step_three.risk_chart).to have_text('53%')
+  expect(js_test? ? step_three.essentials_js : step_three.essentials).to have_text('53%')
+  expect(js_test? ? step_three.essentials_js : step_three.essentials).to have_text('£3,206.21')
+  expect(js_test? ? step_three.total_leftover_js : step_three.total_leftover).to have_text('47%')
+  expect(js_test? ? step_three.total_leftover_js : step_three.total_leftover).to have_text('£2,793.79')
+  expect(js_test? ? step_three.remaining_js : step_three.remaining).to have_text('£1,893.79')
+  expect(js_test? ? step_three.what_if_changes_js : step_three.what_if_changes).to have_text(js_test? ? '3%' : '2%')
+  expect(js_test? ? step_three.what_if_changes_js : step_three.what_if_changes).to have_text(js_test? ? '£2,318.20' : '£2,103.35')
+  expect(js_test? ? step_three.what_if_remaining_js : step_three.what_if_remaining).to have_text(js_test? ? '£1,281.80' : '£1,496.65')
```

Comment:
Well, this requires duplication of expected values, and also adds a level of indirection by putting them behind another method call. I'd be ok with an `if js_test?` inside of the step, with all the expectations there, but at the same time I'm not sure it's a massive improvement just because of the duplication.